lumicks/pylake/kymo.py

```diff
@@ -150,6 +150,87 @@ def set_aspect_ratio(axis, ar):
         set_aspect_ratio(ax1, aspect_ratio)
         set_aspect_ratio(ax2, aspect_ratio)
 
+    def _histogram_rows(self, image, pixels_per_bin, pixel_width):
+        bin_width = pixels_per_bin * pixel_width # in physical units
+        n_rows = image.shape[0]
+        if pixels_per_bin > n_rows:
+            raise ValueError("bin size is larger than the available pixels")
+
+        n_bins = n_rows // pixels_per_bin
+        remainder = n_rows % pixels_per_bin
+        if remainder != 0:
+            warnings.warn(f"{n_rows} pixels is not divisible by {n_bins} bins, skipping last {remainder} pixels")
+        image = image[:(n_bins * pixels_per_bin)].reshape((n_bins, -1))
+
+        counts = image.sum(axis=1)
+        edges = np.arange(n_bins) * bin_width
+        return edges, counts, bin_width
```

Comment:
This function doesn't use self, so it could just be a free function, which would also make it more testable.

Would be nice to have a test for both divisible summation counts and non-divisible with a tiny image where you know what needs to come out.

Nicely vectorized btw!